alunduil/template.py
# Add vale as a devcontainer feature

This references the issue about Vale.

**Problem**

Vale is installed but not found by `postCreateCommand`.

```
 - Installing vale (3.4.1.0)

Installing the current project: templatise (2.2.5)
/bin/sh: 1: vale: not found
[50042 ms] postCreateCommand failed with exit code 127. Skipping any further user-provided commands.
```

**Diagnosis**

Vale in `pyproject.toml` downloads a file but doesn't install it in `PATH`.

```
vscode ➜ /workspaces/template.py (main) $ find / -name vale
/workspaces/template.py/.venv/lib/python3.10/site-packages/vale
/workspaces/template.py/.venv/bin/vale

vscode ➜ /workspaces/template.py (main) $ echo $PATH
/vscode/vscode-server/bin/linux-x64/89de5a8d4d6205e5b11647eb6a74844ca23d2573/bin/remote-cli:/usr/local/python/current/bin:/usr/local/py-utils/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/vscode/.local/bin
```

**Solution**

I added Vale as a feature to `devcontainer.json`. This solves the installation problem and is a better location as it's a feature of devcontainer (rather than a dependency of whatever project someone starts using this template).

**Alternative considered**

Another approach would've been adding `/workspaces/template.py/.venv/bin/vale` to `PATH` in `devcontainer.json`. I don't think this is nearly as clean as a solution as the one I went with.

**How to test**

Create a DevContainer using this pull request and confirm that Vale is installed and `postCreateCommand` runs without errors.
